Code/GraphMol/SmilesParse/smarts.yy

```diff
@@ -162,6 +162,13 @@ ATOM_OPEN_TOKEN bad_atom_def
 mol: atomd {
   int sz     = molList->size();
   molList->resize( sz + 1);
+
+  // The first atom may require a chirality inversion if it has
+  // an implicit H. Mark it to be checked after the mol has been parsed
+  if(sz == 0 && $1->getChiralTag() != Atom::CHI_UNSPECIFIED) {
```

Comment:
I think this is problematic for SMARTS.
The SMILES `[C@H](F)(Cl)C` may need an inversion because the H atom is actually there in the molecule and "participating in defining the chirality".
The SMARTS `[C@H](F)(Cl)C`, on the other hand, doesn't actually have an H affecting the chirality; it has a query that there's an H attached to the C.

Reply:
I see. I misinterpreted the H in the SMARTS, and this actually isn't an issue. I'll comment on #2568 and close it.